### DISCUSSION TITLE
Fetching from enterprise org with Ruby

### DISCUSSION THREAD

**Discussion by zlasa** [octokit/octokit.rb#1504] [Q&A]
Hello, 

I have a console ruby script that has been fetching data from my legacy org. The client has been initialized like this

` @client ||= Octokit::Client.new(:netrc => true).tap {|c| c.auto_paginate = true }`

We are in the process of migrating to an enterprise org that has been configured with SAML. When I try to use the script to get productivity metrics for my team I run into 

`Error ... Resource protected by organization SAML enforcement. You must grant your Personal Access token access to an organization within this business. // See: https://docs.github.com/articles/authenticating-to-a-github-organization-with-saml-single-sign-on/`

Do I need to change the way I am initializing my client? What other configuration do I need to get my ruby script running?

**Comment by kfcampbell** [comment]
@zlasa have you authorized the PAT you're using with the org? There's a menu on https://github.com/settings/tokens that looks like this:


![image](https://user-images.githubusercontent.com/9327688/201212431-ea18d38f-fadb-42b5-b845-5e3ee41766a4.png)

The authorization step is required every time the token is regenerated, for what it's worth. I frequently forget to authorize and receive similar errors when I roll my personal tokens.
